compiler.ts

```diff
@@ -291,10 +315,32 @@ function codeGenExpr(expr : Expr<Type>, env: GlobalEnv) : Array<string> {
   }
 }
 
-function codeGenLiteral(literal : Literal) : Array<string> {
+function allocateStringMemory(string_val:string, env: GlobalEnv) : Array<string>{
+  var newOffset = env.offset;
```

Comment:
I'm concerned that this offset won't update dynamically with allocations. In our object allocations, we read the offset from the heap using a load from the first word of our memory array. Don't we need to also do that for strings so we don't overwrite earlier memory?

Reply:
(For example, if we allocate string memory in a loop, there's no way the compiler can in general know how much memory will be allocated total at runtime, so a static `offset` won't work.)

Reply:
Thanks for this suggestion. I will look into how we are doing object allocations and do the same for strings.